Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Spotify.cpp

```diff
@@ -0,0 +1,241 @@
+    FSong Song4(3, 59, "Gorillas", "New Genious (Brother)");
+
+    FPlayList Playlist2("Gorillas");
+    Playlist2.bAddSong(Song1);
+    Playlist2.bAddSong(Song2);
+    Playlist2.bAddSong(Song3);
+    Playlist2.bAddSong(Song4);
+
+    //Third Playlist
+    FSong SongFour(3, 12, "Nirvana", "About a Girl");;
+    FSong SongFive(2, 40, "Nirvana", "Re-Hash");
+
+    FPlayList Playlist3("Album Nirvana");
+    Playlist3.bAddSong(SongFour);
+    Playlist3.bAddSong(SongFive);
+
+
+    //PlaylistContainer
+    
+    PlayListContainer.AddPlaylist(Playlist1);
+    PlayListContainer.AddPlaylist(Playlist2);
+    PlayListContainer.AddPlaylist(Playlist3);
+}
+FSpotify::FSpotify()
+{
+    ActiveCommand = MenuCommand::MainMenu;
+}
+void FSpotify::RunApp()
+{
+    while (ActiveCommand != MenuCommand::Exit)
+    {
+        switch (ActiveCommand)
+        {
+        case FSpotify::MenuCommand::MainMenu:
+            OpenMainMenu();
+            break;
+        case FSpotify::MenuCommand::CreatePlayList:
+            OpenCreatePlayListMenu();
+            break;
+        case FSpotify::MenuCommand::BrowsePlayLists:
+            OpenBrowsePlayListMenu();
+            break;
+        case FSpotify::MenuCommand::Exit:
+            break;
+        default:
+            break;
+        }
+
+    }
+
+}
+
+
+void FSpotify::OpenMainMenu()
+{
+    do {
+        system("CLS");
+
+        std::cout << "Welcome to Spotify!What would you like to do?" << std::endl;
+        std::cout << "1. Create a PlayList" << std::endl;
+        std::cout << "2. Browse a PlayList" << std::endl;
+        std::cout << "3. Exit" << std::endl;
+        int OptionChoice;
+        std::cin >> OptionChoice;
+        while (!std::cin.good() || OptionChoice < 1 || OptionChoice > 3)
+        {
+            std::cin.clear();
+            std::cin.ignore(1000, '\n');
+            std::cout << "Invalid input,Please enter Your new choice or selection" << std::endl;
+            std::cin >> OptionChoice;
+        }
+        ActiveCommand = (MenuCommand)OptionChoice;
+        system("CLS");
+    } while (!std::cin.good());
+}
+
+
+void FSpotify::OpenCreatePlayListMenu()
+{
+    std::cin.clear();
+    std::cin.ignore(1000, '\n');
+    system("CLS");
+    int AddSongs;
+
+    std::cout << "Creating a new Playlist Fantastic, Please enter a name for Your Playlist: " << std::endl;
+    std::string PlayListName;
+    std::getline(std::cin, PlayListName);
+    while (!std::cin.good() || PlayListName == "" || std::cin.peek() == '\n')
+    {
+        std::cout << "Invalid input, please enter Your playlist name again: " << std::endl;
+        std::getline(std::cin, PlayListName);
+    }
+    FPlayList ANewPlayList(PlayListName);
+    do
+    {
+        system("CLS");
+        std::cout << "Well, now add a song to " << PlayListName << " Playlist" << std::endl;
+        std::cout << "Now,Follow the instructions" << std::endl;
+
+        int EntryMinutesDuration;
```

Comment:
Do you really need to declare all these variables here? Why not declare them closer to where they are actually defined?

Reply:
reubicated